### DISCUSSION TITLE
Support Nested Multi-line Comments

### DISCUSSION THREAD

**Discussion by hannah23280** [dotnet/csharplang#6583] [Language Ideas] [answered]
Hi,
  I would like to suggest supporting multiline comment within multiline comment, similar to that of Kotlin language.   It is very useful for developer to comment a large chunk of code, where inside contain existing multiline comment(s). Such as shown below

```
/*
    /*
        Testing 1
        Testing 2
    */

    Testing 3
    Testing 4

     /*
        Testing 5
        Testing 6
    */


*/
```

**Comment by HaloFour** [comment, marked as answer]
I feel like this could result in a subtle reinterpretation of existing code.  The following is legal today:

```cs
/*
    /*
        Testing 1
        Testing 2
    */

    Testing 3
    Testing 4

     /*
        Testing 5
        Testing 6
    */


// */
```

**Reply by hannah23280** [reply]
> I feel like this could result in a subtle reinterpretation of existing code. The following is legal today:
> 
> ```cs
> /*
>     /*
>         Testing 1
>         Testing 2
>     */
> 
>     Testing 3
>     Testing 4
> 
>      /*
>         Testing 5
>         Testing 6
>     */
> 
> 
> // */
> ```

U are genius!  I never know could achieve the same thing in this way.  I try apply to Java as well, which also don't support nested comment, and it works too !!!!! Thanks.

**Reply by rbwhitaker** [reply]
Yeah, this would be a breaking change. Not that breaking changes can't ever be done, but the bar for justification is *way* higher. You could do what has been done for some language features and make it configurable in the .csproj file, I suppose.

**Comment by hannah23280** [comment]
But nevertheless, I still do hope Csharp is able to support nested multiline comment without the need to add // as workaround.   Cos if Kotlin (a Java-based language) can do that, I believe Csharp should be able too.

**Reply by 333fred** [reply]
As shown by HaloFour, doing so would change the meaning of existing code. This doesn't feel like it's worthy of a breaking change for.

**Comment by PathogenDavid** [comment]
Instead of using comments to disable code you can use conditional compilation, which supports nesting and can toggled easily:

```csharp
#if false
    /*
        Testing 1
        Testing 2
    */

    Testing 3
    Testing 4

     /*
        Testing 5
        Testing 6
    */


#endif
```

**Reply by hannah23280** [reply]
Thanks for the alternative solution. IMO,  typing #if false and toggling to enable it seems more "cumbersome" than just simply put /* to do the job.

**Comment by CyrusNajmabadi** [comment]
I just use VS and do `ctrl k-c` to comment, and `ctrl k-u` to uncomment.  Works great and doesn't require nesting.

**Comment by mattwar** [comment]
I print out the code and cut out the offending bits.
